### PR TITLE
SAN-48 Refactor the PPS baseline web service based on LFP 

### PR DESCRIPTION
**Pull request by klai-ch**
https://companieshouse.atlassian.net/browse/SAN-48

Refactor the PPS baseline web service based on LFP

- For the URI and service call, still keep LFP related
- For some of the web link (e.g. survey), still keep lfp
- For account name for bank transfer, Late Filing is kept

### CODE REVIEW

File: Makefile
Comment (dgordon-ch) [inline Makefile:1]: Update all instances of pps-pay-web to penalty-payment-web per renaming.

File: src/main/java/uk/gov/companieshouse/web/pps/security/WebSecurity.java
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/security/WebSecurity.java:44]: General comment, this is good, removes the weird acronym camel case.

File: src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java
Comment (dgordon-ch) [inline src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java:7]: Already mentioned, not sure if we rename this PayablePenaltyService and remove the Payment bit. Any input is appreciated.

File: start.sh
Comment (dgordon-ch) [inline start.sh:10]: penalty-payment-web

File: src/main/resources/templates/fragments/ppsGlobalErrors.html
Comment (dgordon-ch) [inline src/main/resources/templates/fragments/ppsGlobalErrors.html:11]: For now, let's not touch the html pages as they'll be covered when we go through the Confirmation Statement and other journeys. Revert all the template changes and leave them out of this for now.

**Review by dgordon-ch — CHANGES_REQUESTED** [review]
General comments;

- Good stuff, covered all the bases, controller/service/impl classes all looking good for most part.
- We need to update any reference of 'pps-pay-web' to 'penalty-payment-web' (await confirm before you action this)
- Remove/revert the HTML changes, they'll be done under the UI change tickets and we're keeping a baseline for testing, so UI elements should remain same.

File: README.md
Comment (dgordon-ch) [inline README.md:4]: Update this to  Penalty-Payment-API

File: src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java
Comment (acallaghan-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java:7]: What about 'PenaltyPaymentService', and remove the 'Payable'? Or even just 'PaymentService' as they are all penalties of some sort.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java
Comment (acallaghan-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java:72]: What about even just reverting this for now and have it addressed in a separate ticket? The codebase will have to be made generic, so it might be worth covering that in a ticket, and we can split out the LFP journey and then make rename these.

File: src/main/resources/templates/fragments/ppsGlobalErrors.html
Comment (klai-ch) [reply src/main/resources/templates/fragments/ppsGlobalErrors.html]: Revert back for the template changes (including the naming of the file part), but one thing is enterPPSDetails is reverted back to enterLFPDetails and the related controller part EnterPPSDetailsController.java (line 51) is also revert back to LFP

File: README.md
Comment (klai-ch) [reply README.md]: Changed from PPS-PAY to Penalty-Payment
And PPS_PAY to Penalty_Payment

File: start.sh
Comment (klai-ch) [reply start.sh]: Changed from pps-pay-web to penalty-payment-web

File: src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java
Comment (klai-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/service/penaltypayment/PayablePenaltyPaymentService.java]: Currently changed to PayablePenaltyService, happy to discuss about that.

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java
Comment (klai-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java]: Currently changed to PayablePenaltyService, happy to discuss about that.

File: Tiltfile.dev
Comment (klai-ch) [reply Tiltfile.dev]: Change to eu-west-2

File: Makefile
Comment (klai-ch) [reply Makefile]: Changed from pps-pay-web to penalty-payment-web

File: README.md
Comment (dgordon-ch) [inline README.md:53]: Note -- This is dependant on https://github.com/companieshouse/ci-pipelines/pull/3793/files being pushed, renaming the pipeline names, etc.
Might be worth keeping this LFP for now, until we get the pipeline changes approved -- there's a cleanup task after that for pay-api too, so we can create new ticket for this additional config refactoring.

File: Tiltfile.dev
Comment (dgordon-ch) [inline Tiltfile.dev:8]: Same as above, might be worth keeping this lfp for now.

File: pom.xml
Comment (dgordon-ch) [inline pom.xml:243]: Yeah same as other comments, let's leave this for follow up config change ticket once the pipelines are renamed. Would've been good to have this sorted by now but lead times and resources...

File: start.sh
Comment (dgordon-ch) [inline start.sh:12]:
```suggestion
    PORT="${PENALTY_PAYMENT_WEB_PORT}"
```

File: terraform/groups/ecs-service/main.tf
Comment (dgordon-ch) [inline terraform/groups/ecs-service/main.tf:56]: Again, might be worth leaving this off until we get the pipeline stuff sorted.

File: terraform/groups/ecs-service/locals.tf
Comment (dgordon-ch) [inline terraform/groups/ecs-service/locals.tf:7]: Revert these terraform changes until we've got pipeline updates sorted...see SAN-50.

File: terraform/groups/ecs-service/variables.tf
Comment (dgordon-ch) [inline terraform/groups/ecs-service/variables.tf:114]: As above, remove tf changes for now.

**Review by dgordon-ch — CHANGES_REQUESTED** [review]
Main things; remove the config changes until we get the pipeline stuff sorted, obviously this is taking a bit longer than anticipated so we'll move the config stuff to its own ticket for next sprint (or after).

File: src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java
Comment (dgordon-ch) [reply src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java]: Think getting that as is now is fine, we can always rename it later but yeah, it's "generic" now at least.

File: README.md
Comment (klai-ch) [reply README.md]: Reverted back to lfp-pay-web as suggested

File: Tiltfile.dev
Comment (klai-ch) [reply Tiltfile.dev]: Revert back to lfp-pay-web as suggested, and keep the eu-west-2

File: start.sh
Comment (klai-ch) [reply start.sh]: Changed to PENALTY_PAYMENT_WEB_PORT as suggested

File: terraform/groups/ecs-service/main.tf
Comment (klai-ch) [reply terraform/groups/ecs-service/main.tf]: Revert back to lfp_pay_web as suggested

File: terraform/groups/ecs-service/locals.tf
Comment (klai-ch) [reply terraform/groups/ecs-service/locals.tf]: Revert back to lfp-pay-web as suggested

File: terraform/groups/ecs-service/variables.tf
Comment (klai-ch) [reply terraform/groups/ecs-service/variables.tf]: Revert back to lfp_pay_web as suggested

**Review by dgordon-ch — APPROVED** [review]

**Review by klyttle-ch — APPROVED** [review]

**Review by acallaghan-ch — APPROVED** [review]

File: README.md
Comment (rbull-ch) [inline README.md:27]: If we're changing this Env variable in the `start.sh` top level file - I think that is intended for other environments.  Will you be raising a PR to change this new Env variable in our local docker please?
* https://github.com/companieshouse/docker-chs-development/blob/feature/austin/sanctions/services/modules/lfp-appeals/pps-pay-web.docker-compose.yaml#L31

File: Tiltfile.dev
Comment (rbull-ch) [inline Tiltfile.dev:8]: Just to confirm CH are moving away from using Tilt (deprecated) locally with preference for just `chs-dev` deployments. 

Mainly wanted to say that the local image for lfp-pay-web is actually still on eu-west-1 development currently which is confusing.

See FYI https://eu-west-2.console.aws.amazon.com/ecr/repositories/private/169942020521/local/lfp-pay-web?region=eu-west-2

File: src/main/resources/ValidationMessages.properties
Comment (rbull-ch) [inline src/main/resources/ValidationMessages.properties:3]: Since we touched these 2 lines, similar to the line above, I think the whitespace can be removed around the '='

File: pom.xml
Comment (rbull-ch) [inline pom.xml:243]: Please can this be `169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/lfp-pay-web:latest`
Looking at our aws profiles `~/.aws/config`, my understanding is that the image should belong to 
`profile development-eu-west-1` instead of `profile shared-services-ecr-eu-west-1-ro]`

File: README.md
Comment (rbull-ch) [reply README.md:27]: Branch created - https://github.com/companieshouse/docker-chs-development/tree/feature/austin/SAN-48-PPS-Web-Rename

Comment (klai-ch) [reply README.md:27]: The PR is created 
https://github.com/companieshouse/docker-chs-development/pull/1497
And the branch is 
https://github.com/companieshouse/docker-chs-development/tree/feature/austin/SAN-48-PPS-Web-Rename-Env

Comment (rbull-ch) [reply README.md:27]: Approved - happy for you to merge https://github.com/companieshouse/docker-chs-development/pull/1497 once the repos are renamed and this PR gets merged.  Thanks

File: pom.xml
Comment (klai-ch) [reply pom.xml]: Updated from the suggestion above

**Review by rbull-ch — APPROVED** [review]

**Pull request merged**
